src/main/java/com/internet/mixshop/controller/AddProductController.java

```diff
@@ -0,0 +1,32 @@
+package com.internet.mixshop.controller;
+
+import com.internet.mixshop.lib.Injector;
+import com.internet.mixshop.model.Product;
+import com.internet.mixshop.service.ProductService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddProductController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.mixshop");
+    private ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/view/product/add-product.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String price = req.getParameter("price");
```

Comment:
What if we receive the empty Strings? :)

Reply:
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

Reply:
ok